### PR TITLE
Fix get tables not escape the wildcard chars in IgniteClient

### PR DESCRIPTION
**Pull request by chenjian2664**
## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Review by ebyhr — COMMENTED** [review]
Please add a test.

File: plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java [hunk 1]
```diff
@@ -95,6 +95,35 @@ protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
         }
     }
 
+    /**
+     * Test that we are correctly escaped the wildcard characters while requesting DataBaseMetaData: getSchemas and getTables with params contains pattern.
```
Comment (ebyhr) [inline plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java:99]: The test method name is enough descriptive. I would remove this comment.

File: plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java [hunk 2]
```diff
@@ -95,6 +95,35 @@ protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
...
+    @Test
+    public void testDatabaseMetadataSearchEscapedWildCardCharacters()
+    {
+        // wildcard characters on schema name
+        assertQuerySucceeds("SHOW TABLES FROM public");
+        assertQueryFails("SHOW TABLES FROM \"publi_\"", ".*Schema 'publi_' does not exist");
+        assertQueryFails("SHOW TABLES FROM \"pu%lic\"", ".*Schema 'pu%lic' does not exist");
+
+        try {
+            assertUpdate("CREATE TABLE abcd (a int, b int, c int) WITH (primary_key = ARRAY['a'])");
```
Comment (ebyhr) [inline plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java:110]: Rename "abcd" to "test_search" or something and add random prefix.

File: plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java [hunk 3]
```diff
@@ -95,6 +95,35 @@ protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
...
+            assertUpdate("CREATE TABLE ab_d (a int, b int, c int) WITH (primary_key = ARRAY['a'])");
+            assertUpdate("CREATE TABLE \"a%d\" (a int, b int, c int) WITH (primary_key = ARRAY['a'])");
```
Comment (ebyhr) [inline plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java:112]: Specify different `primary_key` in each table definition like below and assert the table definition. 
* abcd → a
* ab_d → b
* \"a%d\" → c

File: plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java [hunk 4]
```diff
@@ -95,6 +95,35 @@ protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
...
+            // wildcard characters on table name
+            assertQuerySucceeds("SHOW CREATE TABLE ab_d");
+            assertQuerySucceeds("SHOW CREATE TABLE \"a%d\"");
+            assertQueryFails("SHOW CREATE TABLE \"a%\"", ".*Table 'ignite.public.a%' does not exist");
+            assertQueryFails("SHOW COLUMNS FROM \"a%\"", ".*Table 'ignite.public.a%' does not exist");
+        }
+        finally {
+            assertUpdate("DROP TABLE abcd");
+            assertUpdate("DROP TABLE ab_d");
+            assertUpdate("DROP TABLE \"a%d\"");
```
Comment (ebyhr) [inline plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java:123]: These tables may not exist if the above CREATE TABLE statement failed. Please move CREATE TABLE to before `try`  or use `DROP TABLE IF EXISTS`.

**Review by ebyhr — APPROVED** [review]

File: plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java (hunk 2 above)
Comment (chenjian2664) [reply plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java:110]: renamed and added a random suffix.

File: plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java [hunk 5]
```diff
@@ -95,6 +96,35 @@ protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
...
+    public void testDatabaseMetadataSearchEscapedWildCardCharacters()
+    {
+        // wildcard characters on schema name
+        assertQuerySucceeds("SHOW TABLES FROM public");
+        assertQueryFails("SHOW TABLES FROM \"publi_\"", ".*Schema 'publi_' does not exist");
+        assertQueryFails("SHOW TABLES FROM \"pu%lic\"", ".*Schema 'pu%lic' does not exist");
+
+        String normalTableName = "testxsearch" + randomNameSuffix();
+        String underscoreTableName = "\"" + "test_search" + randomNameSuffix() + "\"";
+        String percentTableName = "\"" + "test%search" + randomNameSuffix() + "\"";
```
Comment (ebyhr) [inline plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java:109]: The same suffix should be reused.

File: plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java [hunk 6]
```diff
@@ -95,6 +96,35 @@ protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
...
+        try {
+            assertUpdate("CREATE TABLE " + normalTableName + "(a int, b int, c int) WITH (primary_key = ARRAY['a'])");
+            assertUpdate("CREATE TABLE " + underscoreTableName + "(a int, b int, c int) WITH (primary_key = ARRAY['b'])");
+            assertUpdate("CREATE TABLE " + percentTableName + " (a int, b int, c int) WITH (primary_key = ARRAY['c'])");
+
+            // wildcard characters on table name
+            assertQuerySucceeds("SHOW CREATE TABLE " + underscoreTableName);
+            assertQuerySucceeds("SHOW CREATE TABLE " + percentTableName);
```
Comment (ebyhr) [inline plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java:117]: Please verify if the result contains the expected `primary_key` table property in 3 tables. e.g. 
```java
assertThat((String) computeScalar("SHOW CREATE TABLE " + x)).contains("primary_key = ARRAY['a'])");
```

**Pull request merged**